File: 프로그래머스연습문제/신규아이디추천.cpp
#include <string>
#include <vector>
#include <algorithm>


using namespace std;

string solution(string new_id) {
    string answer = "";
    
    for(int i = 0; i < new_id.size(); i++){
        new_id[i] = tolower(new_id[i]);
    }
    
    for (int i = 0; i < new_id.size(); i++){
        if(new_id[i] <= 'z' && new_id[i] >= 'a') continue;
        else if(new_id[i] >= '0' && new_id[i] <='9' )continue;
        else if(new_id[i] == '-' || new_id[i] == '_' || new_id[i] == '.')continue;
        else new_id[i] = ' ';
    }
    new_id.erase(remove(new_id.begin(),new_id.end(),' '), new_id.end());
    

    
    int index = 0;
    while(index < new_id.size()){
        if(new_id[index] == new_id[index+1] && new_id[index] == '.'){
            new_id = new_id.substr(0,index)+new_id.substr(index+1,new_id.size()-index-1);
        }
        else index++;
    }
    
    while(1){
        if(new_id[0] == '.')new_id = new_id.substr(1,new_id.size()-1);
        
        if(new_id[new_id.size()-1] == '.') new_id = new_id.substr(0,new_id.size()-1);
        
        if(new_id[0] != '.' && new_id[new_id.size() - 1] != '.')break;
    }
    
    if (new_id.size() == 0)new_id += 'a';
    
    if(new_id.size() >= 16)new_id = new_id.substr(0,15);
    while(new_id[new_id.size()-1] =='.')new_id = new_id.substr(0,new_id.size()-1);
    
    while (new_id.size() <=2 )new_id = new_id + new_id[new_id.size()-1];
    
    answer = new_id;
    
    return answer;
}

File: 카카오/2021 카카오 블라인드/신규 아이디 추천.py
def solution(new_id):
    answer = ''
    index = 0
    
    #step 1
    word = new_id.lower()
    
    #step 2
    for i in range(len(word)):
        if word[i].isalnum() or word[i] == '-' or word[i] == '_' or word[i] == '.':
            answer += word[i]
    
    word = answer
    answer = ''
    
    #step 3
    subword = ''
    while(index < len(word)):
        if word[index] == '.':
            subword = '.' 
        else :
            answer += subword + word[index]
            subword = ''
            
        index += 1
    if len(subword) != 0:
        answer += subword
        
    word = answer
    answer = ''
        
    #step 4
    if word.startswith('.') == True:
        word = word[1:len(word)]
    if word.endswith('.') == True:
        word = word[0:len(word) - 1]
        
    if len(word) == 0:
        word += 'a'
    #step 6
    if len(word) >= 16:
        word = word[:15]
    if word.endswith('.') == True:
        word = word[:len(word) - 1]
        
    #step 7
    while len(word) <= 2:
        word += word[len(word) - 1]
    
    answer = word
    return answer
